### PR TITLE
feat: support label/selector change of employee/employer in resource …

### PR DESCRIPTION
**Pull request by WeichengWang1**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 
re #73

#### 2. What is the scope of this PR (e.g. component or file name):
resource consist

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/controllers/resourceconsist/consister.go [hunk 1]
```diff
@@ -201,13 +202,60 @@ func (r *Consist) syncEmployees(ctx context.Context, employer client.Object, exp
 	toAddLifecycleFlzEmployees, toDeleteLifecycleFlzEmployees := r.getToAddDeleteLifecycleFlzEmployees(
```
Comment (wu8685) [inline pkg/controllers/resourceconsist/consister.go:202]: Should we use the `names` recorded on annotation as the `currentEmployees`? Not the `names` recorded in backend service.

File: pkg/controllers/resourceconsist/consister.go [hunk 2]
```diff
@@ -201,13 +202,60 @@ func (r *Consist) syncEmployees(ctx context.Context, employer client.Object, exp
...
 		succCreate, succDelete, succUpdate, toCudEmployees.Unchanged)
 
+	recordedNotSelected := make(map[string]bool)
```
[the remark this answers is not in the document]
Comment (WeichengWang1) [reply pkg/controllers/resourceconsist/consister.go:205]: should be removed

File: pkg/controllers/resourceconsist/consister.go (hunk 1 above)
Comment (WeichengWang1) [reply pkg/controllers/resourceconsist/consister.go:202]: names used here is for add/remove finalizers for pods, names used in diff is employeeId(names) in backend service

File: pkg/controllers/resourceconsist/consister.go [hunk 3]
```diff
@@ -201,13 +202,60 @@ func (r *Consist) syncEmployees(ctx context.Context, employer client.Object, exp
...
 
+	if needRecordEmployees {
+		needUpdate := false
+		if employer.GetAnnotations()[lifecycleFinalizerRecordedAnnoKey] == "" {
+			if len(toAddLifecycleFlzEmployees) != 0 {
+				needUpdate = true
+			}
+		} else {
+			recordedEmployees := strings.Split(employer.GetAnnotations()[lifecycleFinalizerRecordedAnnoKey], ",")
+			if !reflect.DeepEqual(recordedEmployees, toAddLifecycleFlzEmployees) {
```
[the remark this answers is not in the document]
Comment (WeichengWang1) [reply pkg/controllers/resourceconsist/consister.go:240]: yes
need to fix

File: pkg/controllers/resourceconsist/types.go [hunk 4]
```diff
@@ -43,10 +47,19 @@ type ReconcileWatchOptions interface {
 	EmployeePredicates() predicate.Funcs
 }
 
+// ReconcileLifecycleOptions defines whether PodOpsLifecycle followed and
+// whether employees' LifecycleFinalizer conditions need to be Record/Erase to employer's anno
+// actually NeedRecordEmployees only needed for those adapters that follow PodOpsLifecycle,
```
Comment (shaofan-hs) [inline pkg/controllers/resourceconsist/types.go:52]: actually -> Actually

**Review by shaofan-hs — APPROVED** [review]

**Pull request merged**